src/sources/soundsourceffmpeg.cpp

```diff
@@ -1159,24 +1159,27 @@ ReadableSampleFrames SoundSourceFFmpeg::readSampleFramesClamped(
                 }
                 const auto decodedFrameCount = m_pavDecodedFrame->nb_samples;
                 DEBUG_ASSERT(decodedFrameCount > 0);
-                auto streamFrameIndex =
+                SINT streamFrameIndex =
                         convertStreamTimeToFrameIndex(
                                 *m_pavStream, m_pavDecodedFrame->pts);
-                // Only audible samples are counted, i.e. any inaudible aka
-                // "priming" samples are not included in nb_samples!
-                // https://bugs.launchpad.net/mixxx/+bug/1934785
-                if (streamFrameIndex < kMinFrameIndex) {
+
+                if (m_avutilVersion >= AV_VERSION_INT(56, 52, 100)) {
+                    // From ffmpeg 4.4 only audible samples are counted, i.e. any inaudible aka
+                    // "priming" samples are not included in nb_samples!
+                    // https://github.com/mixxxdj/mixxx/issues/10464
+                    if (streamFrameIndex < kMinFrameIndex) {
 #if VERBOSE_DEBUG_LOG
-                    const auto inaudibleFrameCountUntilStartOfStream =
-                            kMinFrameIndex - streamFrameIndex;
-                    kLogger.debug()
-                            << "Skipping"
-                            << inaudibleFrameCountUntilStartOfStream
-                            << "inaudible sample frames before the start of the stream";
+                        const auto inaudibleFrameCountUntilStartOfStream =
+                                kMinFrameIndex - streamFrameIndex;
+                        kLogger.debug()
+                                << "Skipping"
+                                << inaudibleFrameCountUntilStartOfStream
+                                << "inaudible sample frames before the start of the stream";
 #endif
-                    streamFrameIndex = kMinFrameIndex;
+                        streamFrameIndex = kMinFrameIndex;
+                    }
                 }
-                DEBUG_ASSERT(streamFrameIndex >= kMinFrameIndex);
```

Comment:
still don't understand why you removed this. Judging by naming this makes sense, why would `streamFrameIndex < kMinFrameIndex` be valid if `kMinFrameIndex` is supposed to denote a minimum?

Reply:
ffmpeg add priming samples before the requested samples start this is the reason why streamFrameIndex can be negative. 

So it turns out that kMinFrameIndex = 0 is badly named. It was also confusing to me. 
I think we should replace it with just 0 for a better readability.

Reply:
Since the frame count is correct before ffmpeg 4.4 the code below deals with that case. The hack of setting the start frame to zero is only required for newer ffmpeg versions. 
Do you have a suggestion to improve the code or comments here?

Reply:
I don't think inlining the constant here improves readability. 
The question I rather have is whether its valid to have `streamFrameIndex < 0` before ffmpeg 4.4?

Reply:
Yes, the assertion will kick in here before 4.4

Reply:
Okay, thanks for clarifying.